src/Interpreters/executeQuery.cpp

```diff
@@ -651,7 +653,10 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 
             elem.client_info = client_info;
 
-            bool log_queries = settings.log_queries && !internal;
+            std::bernoulli_distribution should_log{
```

Comment:
You've made it only for query start.
There is also query finish event.

And it needs to be consistent: if query is logged - it applied both for start and finish.

Reply:
Is not this exact bool variable `log_queries` captured in finish callback (line 730) and then used to determine whether to write query to log (line 797)?